web/client/components/geostory/common/enhancers/map.jsx

```diff
@@ -40,8 +40,11 @@ export default compose(
             const resource = find(resources, { id: resourceId }) || {};
             const baseMap = omit(resource.data, ['context']);
             const isLegacyGeostory = (map?.layers || [])?.indexOf(null) !== -1 || (map?.groups || [])?.indexOf(null) !== -1;
+            // 'layersGroupsHasEmptyItems' is a flag that indicates if the geostory 'map' object in section includes empty layers/groups
+            // the layers/groups array may include empty items in case change layer properties e.g: opacity,title ...etc
+            const layersGroupsHasEmptyItems = (map?.layers || []).includes(undefined) || (map?.groups || []).includes(undefined);
```

Comment:
This implementation seems to solve the problem but it introducing a new set of checks for a specific configuration caused from an inconsistency on the way we update layers/groups in the settings panel.

I would like to propose a change inside the [MapEditor](https://github.com/geosolutions-it/MapStore2/blob/master/web/client/components/geostory/common/MapEditor.jsx#L105) instead. The `onChange` of the Editor component should update the layers or groups key of the map instead to target a specific index of the list of layers/groups. Something like the following code:

```js
<Editor
    closeNodeEditor={closeNodeEditor}
    editNode={editNode}
    map={map}
    onChange={(path, value) => {
        const config = set({ map: { ...map } }, path, value);
        const { layers, groups } = config?.map || {};
        if (path.includes('map.layers[')) {
            onChangeMap('layers', layers);
        } else {
            onChangeMap('groups', groups);
        }
    }}/>
```
By modifying the way we update the layers/groups keys of the map on the setting we are going to align it to the behaviour of the TOC. If we are introducing this change we don't need the `layersGroupsHasEmptyItems` introduced by this PR